src/parser.rs

```diff
@@ -6407,6 +6426,52 @@ impl<'a> Parser<'a> {
                 } else {
                     expr
                 };
+                let mut range: Option<Value> = None;
+                let mut fill: Option<Ident> = None;
+                for _ in 0..2 {
+                    if self.parse_keyword(Keyword::RANGE) {
+                        range = Some(self.parse_value()?);
+                    }
+                    if self.parse_keyword(Keyword::FILL) {
+                        fill = Some(self.parse_identifier()?);
+                    }
+                }
```

Comment:
Unlike the tailing `ALIGN` and `FILL`, `RANGE ... [FILL ...]` should have the fixed ordering. I.e., `FILL ... RANGE ...` is invalid.